packages/flutter/test/cupertino/text_field_test.dart

```diff
@@ -749,6 +749,45 @@ void main() {
     },
   );
 
+  testWidgets(
+    'clear button tapped alse call onChanged when text not empty',
+        (WidgetTester tester) async {
+      String value = '';
+      final TextEditingController controller = TextEditingController();
+      await tester.pumpWidget(
+        CupertinoApp(
+            home: StatefulBuilder(
+                builder: (BuildContext context, StateSetter setState) {
+                  return Center(
+                    child: CupertinoTextField(
+                      controller: controller,
+                      placeholder: 'placeholder',
+                      onChanged: (String newValue) {
+                        setState(() {
```

Comment:
You don't need the set state here since you're just setting a variable outside of any widgets.